### PR TITLE
feat(DTFS2-5945): Amendment change leadUnderwriterId to leadUnderwriter object

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Changes made:
* updated leadUnderwriterId so it is stored as object containing  _id and first and last name
* updated tests

### CODE REVIEW

File: trade-finance-manager-ui/server/controllers/case/amendments/leadUnderwriter.controller.js [hunk 1]
```diff
@@ -88,16 +90,30 @@ const getAssignAmendmentLeadUnderwriter = async (req, res) => {
...
+    const { assignedTo: assignedToValue } = req.body;
+    // assignedToValue is only the id so have to get the user's name from api call
+    const user = await api.getUser(assignedToValue);
 
-  const { assignedTo: assignedToValue } = req.body;
-  const update = { leadUnderwriterId: assignedToValue };
+    // if no user.firstName or lastName, then values set to null
+    const updateUser = {
+      _id: assignedToValue,
+      firstName: user.firstName ? user.firstName : null,
```
Comment (IgnatG) [inline trade-finance-manager-ui/server/controllers/case/amendments/leadUnderwriter.controller.js:107]: can we use something like:
`firstName: user.firstName ?? null`
?

File: trade-finance-manager-ui/server/controllers/case/amendments/leadUnderwriter.controller.js [hunk 2]
```diff
@@ -88,16 +90,30 @@ const getAssignAmendmentLeadUnderwriter = async (req, res) => {
...
+      lastName: user.lastName ? user.lastName : null,
```
Comment (abhi-markan) [inline trade-finance-manager-ui/server/controllers/case/amendments/leadUnderwriter.controller.js:108]: I would suggest usage of nullish coalescing for the lastname as well please.

File: trade-finance-manager-ui/server/controllers/case/amendments/leadUnderwriter.controller.js [hunk 3]
```diff
@@ -88,16 +90,30 @@ const getAssignAmendmentLeadUnderwriter = async (req, res) => {
...
+    }
+
+    const { assignedTo: assignedToValue } = req.body;
+    // assignedToValue is only the id so have to get the user's name from api call
+    const user = await api.getUser(assignedToValue);
 
-  const { assignedTo: assignedToValue } = req.body;
-  const update = { leadUnderwriterId: assignedToValue };
+    // if no user.firstName or lastName, then values set to null
+    const updateUser = {
```
Comment (abhi-markan) [inline trade-finance-manager-ui/server/controllers/case/amendments/leadUnderwriter.controller.js:105]: Shall we rename this variable to `leadUnderwriter` (if no conflicts)?

**Review by abhi-markan — CHANGES_REQUESTED** [review]

File: trade-finance-manager-ui/server/controllers/case/amendments/leadUnderwriter.controller.js (hunk 1 above)
Comment (Zainzzkk) [reply trade-finance-manager-ui/server/controllers/case/amendments/leadUnderwriter.controller.js:107]: done

File: trade-finance-manager-ui/server/controllers/case/amendments/leadUnderwriter.controller.js (hunk 2 above)
Comment (Zainzzkk) [reply trade-finance-manager-ui/server/controllers/case/amendments/leadUnderwriter.controller.js:108]: done

File: trade-finance-manager-ui/server/controllers/case/amendments/leadUnderwriter.controller.js (hunk 3 above)
Comment (Zainzzkk) [reply trade-finance-manager-ui/server/controllers/case/amendments/leadUnderwriter.controller.js:105]: done

**Review by abhi-markan — APPROVED** [review]
Nice work

**Pull request merged**